src/main/java/com/project/playlist/service/impl/PlaylistVideoServiceImpl.java

```diff
@@ -0,0 +1,121 @@
+package com.project.playlist.service.impl;
+
+
+import com.project.playlist.model.Playlist;
+import com.project.playlist.model.PlaylistVideo;
+import com.project.playlist.model.Video;
+import com.project.playlist.repository.PlaylistVideoRepository;
+import com.project.playlist.service.PlaylistService;
+import com.project.playlist.service.PlaylistVideoService;
+import com.project.playlist.service.VideoService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Service
+@Transactional
+public class PlaylistVideoServiceImpl implements PlaylistVideoService {
+    @Autowired
+    private VideoService videoService;
+    @Autowired
+    private PlaylistService playlistService;
+    @Autowired
+    private PlaylistVideoRepository playlistVideoRepository;
+
+    @Override
+    public void addVideoToPlaylist(Long playlistId, Long videoId) {
+        Playlist playlist = playlistService.getPlaylistById(playlistId);
+        Video video = videoService.getVideoById(videoId);
+
+        List<PlaylistVideo> playlistVideoList = playlist.getPlaylistVideos();
+        if (playlistVideoList.stream().anyMatch(pv -> pv.getVideo().getId().equals(videoId))) {
+            System.out.println("Video with id " + videoId + " already exists in playlist.");
```

Comment:
Remove `System.out.println()` and create new `VideoAlreadyExists` exception which you will throw here